### PR TITLE
Add support for retrieving game server Steam Id (bug 6404).

### PR DESCRIPTION
**Pull request by psychonic**
This has been requested a few times and isn't too bad to add.

It could have been done simpler, but this leaves a segue to move more natives to the logic bin.

### CODE REVIEW

File: core/logic/smn_halflife.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+static cell_t GetServerAccountId(IPluginContext *pContext, const cell_t *params)
+{
+	return (cell_t)(gamehelpers->GetServerSteamId64() & 0xFFFFFFFF);
+}
+
+REGISTER_NATIVES(halflifeNatives)
+{
+	{ "GetServerAuthId",			GetServerAuthId },
+	{ "GetServerSteamAccountId",	GetServerAccountId },
```
Comment (KyleSanderson) [inline core/logic/smn_halflife.cpp]: Should be similar to GetSteamAccountID so it doesn't drive people crazy.

Comment (psychonic) [inline core/logic/smn_halflife.cpp]: The existing GetPlayerAccountID is the exception, not the rule. We use "Id" everywhere else in the API. (It should actually be GetClientAccountId - it's just a bit late for that now).

**Comment by psychonic** [comment]
As @KyleSanderson (?) pointed out, this didn't actually work at runtime on Linux (or Mac) for the engine on which we needed to call the steam_api function directly to get the id.

This has now been fixed.

**Comment by KyleSanderson** [comment]
:shipit:

**Pull request merged**